integration/examples/bazel/skaffold.yaml

```diff
@@ -5,3 +5,7 @@ build:
   - image: skaffold-bazel
     bazel:
       target: //:skaffold_example.tar
+deploy:
+  kubectl:
+    manifests:
+      - k8s/*
```

Comment:
Since Skaffold adds this by default, should it be defaulted for the required-config case too?

Reply:
> Why is this important? Is this exercising new functionality that would otherwise be untested? Is there functionality in our examples that others may want to reuse?
> 
> This change will add more time to our tests, and I'd like to ensure there's a benefit.

So the main point is to test that new features don't incorrectly assume `working directory`. For instance, https://github.com/GoogleContainerTools/skaffold/pull/5451 was adding new config fields that reference to files on the filesystem. It'd be a manual activity every time to make sure that `skaffold:"filepath"` tags are added or that there's no assumption about a global `workDir`.  WDYT?

Reply:
> Since Skaffold adds this by default, should it be defaulted for the required-config case too?

In a single config it's plausible to add a default deploy stanza. For multiple configs, there might be build-only configs (see our [microservices](https://github.com/GoogleContainerTools/skaffold/blob/master/examples/multi-config-microservices/base/skaffold.yaml) example) which aren't deployed. WDYT?

Reply:
Not all of our examples are included in this TestRun.  This seems an expensive way to catch fields missing the filepath tag.

I think you could have a test that walks all of the `skaffold.yaml` found under `integration/` and:
  1. loads the skaffold.yaml, 
  2. walks the fields looking for fields *without* the `filepath` tag
  3. checks to see if a file by the value exists under the path of the skaffold.yaml and if so, throws an error.

(We'll need special casing for file references within artifacts.)

Reply:
We don't do any validations with our `filepath` tag so I'm not sure about your suggestion. To clarify about my intent, consider some feature implemented whose definition in `skaffold.yaml` looks like:
```yaml
fooFeature:
  path: ./bar.sh
```
 and the `config.go` equivalent struct is:
 ```go
 type FooFeature  struct {
  Path string `yaml:"path"`   // missing `filepath` tag
}
 ```

and in some `foo.go` file it's logic is:
```go
wd, _ := os.Getwd()
fooPath := filepath.Join(wd, config.FooFeature.Path)
```
This `skaffold.yaml` cannot be used as a dependency from other `skaffold.yaml` files in different directories since the value of `fooFeature.path` changes with `wd`.

If the above struct definition had the `filepath` tag included:

```go

 type FooFeature  struct {
  Path string `yaml:"path" skaffold:"filepath"` 
}
```

then the path would be always relative to the `skaffold.yaml` file it's defined in. This way the same file can be imported as a dependency from any other directory.

---

If it's too expensive to run it on the list of sample projects how about running it only on newly added projects (diff between `integration/examples` and `examples`) ? That'd add an occasional 1-2 tests.